qcodes/instrument_drivers/stanford_research/SR830_channels.py

```diff
@@ -0,0 +1,602 @@
+from functools import partial
+import numpy as np
+
+from qcodes import VisaInstrument
+from qcodes.instrument.parameter import ArrayParameter
+from qcodes.utils.validators import Numbers, Ints, Enum, Strings
+from qcodes.instrument.channel import InstrumentChannel, ChannelList
+
+
+class SR830AuxChannel(InstrumentChannel):  # We see that InstrumentChannel, ChannelList are something of a misnomer
+    def __init__(self, parent, name, aux_number):
+        super().__init__(parent, name)
+
+        self.add_parameter('aux_in',
+                           label='Aux input {}'.format(aux_number),
+                           get_cmd='OAUX? {}'.format(aux_number),
+                           get_parser=float,
+                           unit='V')
+
+        self.add_parameter('aux_out',
+                           label='Aux output {}'.format(aux_number),
+                           get_cmd='AUXV? {}'.format(aux_number),
+                           get_parser=float,
+                           set_cmd='AUXV {0}, {{}}'.format(aux_number),
+                           unit='V')
+
+
+class SR830Channel(InstrumentChannel):
+    def __init__(self, parent, name, channel_number):
+        super().__init__(parent, name)
+        # detailed validation and mapping performed in set/get functions. Note that we only have
+        # two channels
+        self.add_parameter(
+            'ratio',
+            label='Channel {} ratio'.format(channel_number),
+            get_cmd=partial(parent._get_ch_ratio, channel_number),
+            set_cmd=partial(parent._set_ch_ratio, channel_number),
+            vals=Strings()
+        )
+
+        self.add_parameter(
+            'display',
+            label='Channel {} display'.format(channel_number),
+            get_cmd=partial(parent._get_ch_display, channel_number),
+            set_cmd=partial(parent._set_ch_display, channel_number),
+            vals=Strings()
+        )
+
+        self.add_parameter(
+            'databuffer',
+            channel=channel_number,
+            parameter_class=ChannelBuffer
+        )
+
+
+class ChannelBuffer(ArrayParameter):
+    """
+    Parameter class for the two channel buffers
+
+    Currently always returns the entire buffer
+    TODO (WilliamHPNielsen): Make it possible to query parts of the buffer.
+    The instrument natively supports this in its TRCL call.
+    """
+
+    def __init__(self, name: str, channel_array: 'SR830Channel', channel: int):
```

Comment:
This will not work. `channel_array` must be renamed to `instrument`, although that is not a very fitting name in this case. We could consider changing that to `parent` in the parameter class.